iothub/device/src/ClientProperties.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// A container for properties retrieved from the service.
+    /// </summary>
+    /// <remarks>
+    /// The <see cref="ClientProperties"/> class is not meant to be constructed by customer code.
+    /// It is intended to be returned fully populated from the client method <see cref="InternalClient.GetClientPropertiesAsync(System.Threading.CancellationToken)"/>.
+    /// </remarks>
+    public class ClientProperties : ClientPropertyCollection
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="ClientProperties"/>.
+        /// This is provided for unit testing purposes only.
+        /// </summary>
+        /// <inheritdoc path="/remarks" cref="ClientProperties" />
+        public ClientProperties()
```

Comment:
Should this be internal if it is for unit testing only?

Reply:
it'll also allow external users to be able to mock this, if they want